### DISCUSSION TITLE
Modifying restarts

### DISCUSSION THREAD

**Discussion by twanverweij** [InteractiveComputerGraphics/SPlisHSPlasH#331] [Q&A]
Dear Jan and team,

After having worked with SPlisHSPlasH for a while now I am in awe of the code base you provide, the possibilities are endless it seems, and seemingly large modifications and optimizations can be implemented without hassle. Many thanks for sharing!

I've managed to implement/tweak the work of Gissler et al. [https://cg.informatik.uni-freiburg.de/publications/2020_SIGGRAPH_snow_v6.pdf](url) to suit my needs for viscoelastic / plastic behaviour in ICSPH. To keep track of deformations however, this method requires keeping track of the deformation gradient which is a Matrix3r.

As the process modeled is a batch process, I would like to restart the simulation multiple times while keeping track of this quantity, adding new batches of particles every cycle. The question is however, what the best route would be. The partio-writer to .bgeo has the limitation of not being able to save anything beyond vectors. I could try a quick and dirty split of my deformation gradient matrix into vectors and making sure the simulation quantities are used after loading the .bgeo as FluidModel. Before I do however, I'm curious to learn if there are any better ways. Ideally usage of the binary state output format appears much cleaner.  

Wouldn't it be better/possible to add particles with a deferred initialization after loading a binary state? Without diving deep,are there any blockers in the I/O architecture of SPlisHSPlasH for this to not work?

Thanks in advance,
Twan

**Comment by janbender** [comment]
You can save the current state using a binary writer. If you want to stort a matrix, just take a look at this example here: 
https://github.com/InteractiveComputerGraphics/SPlisHSPlasH/blob/08f9fc8fad5efb5e346b8b6be4bfaa42be3845a2/SPlisHSPlasH/Elasticity/Elasticity_Peer2018.cpp#L596
